tests/python/VultureBearTest.py

```diff
@@ -42,85 +52,64 @@ def prep_file():
 
             return list(self.uut.run())
 
+    def verify_results(self, test_case, expected):
+        detected = dict((item.message, (item.affected_code[0].start.line,
+                                        item.affected_code[0].end.line,
+                                        item.confidence))
+                        for item in self.get_results(test_case))
+        self.assertEqual(detected, expected)
+
     def test_used_variable(self):
-        good_file = """
-        x = 2
-        print(x)
-        """
-        self.assertEqual(len(self.get_results(good_file)), 0)
+        self.verify_results(load_testfile('used_variable.py'), {})
 
     def test_unused_variable(self):
-        bad_file = """
-        b = 10
-        a = 12
-        print(a)
-        """
-        self.assertEqual(len(self.get_results(bad_file)), 1)
-
-    def test_unused_parameter(self):
-        bad_file = """
-        def test(a):
-            return 1
-
-        test(1)
-        """
-        self.assertEqual(len(self.get_results(bad_file)), 1)
-
-    def test_unused_function(self):
-        bad_file = """
-        def test(a):
-            return a
-        """
-        self.assertEqual(len(self.get_results(bad_file)), 1)
-
-    def test_import_confidence(self):
-        bad_file = """
-        from os import *
-        import subprocess
-        """
-        result = self.get_results(bad_file)
-        self.assertEqual(len(result), 1)
-        self.assertEqual(result[0].confidence, 95)
-
-    def test_class_confidence(self):
-        bad_file = """
-        class Name:
-
-            def __init__(self, name):
-                self.name = name
-                self.sur_name = 'Something'
-
-        """
-        result = self.get_results(bad_file)
-        self.assertEqual(len(result), 2)
-        self.assertEqual(result[0].confidence, 70)
-
-    def test_func_confidence(self):
-        bad_file = """
-        def hello(name):
-            print('Hello World')
-        """
-        result = self.get_results(bad_file)
-        self.assertEqual(len(result), 2)
-        self.assertEqual(result[0].confidence, 70)
-
-    def test_prop_confidence(self):
-        bad_file = """
-        class Bar(object):
-            @property
-            def prop(self):
-                pass
-
-        c = Bar()
-        """
-        result = self.get_results(bad_file)
-        self.assertEqual(len(result), 2)
-        self.assertEqual(result[0].confidence, 70)
-
-    def test_var_confidence(self):
-        bad_file = """
-        name = 'Foo Bar'
-        """
-        result = self.get_results(bad_file)
-        self.assertEqual(len(result), 1)
-        self.assertEqual(result[0].confidence, 70)
+        code = load_testfile('unused_variable.py')
```

Comment:
I'd pass the filename to `self.verify_results()`.

Reply:
Done! :+1: